state/indexer/block/kv/kv.go

```diff
@@ -95,11 +100,25 @@ func (idx *BlockerIndexer) Search(ctx context.Context, q *query.Query) ([]int64,
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse query conditions: %w", err)
 	}
+	// conditions to skip because they're handled before "everything else"
+	skipIndexes := make([]int, 0)
+
+	// Check whether we want to return heights where the conditions are true
+	// within the same event or it does not matter
+	matchEvents, matchEventIdx := lookForMatchEvent(conditions)
 
+	if matchEventIdx != -1 {
+		skipIndexes = append(skipIndexes, matchEventIdx)
+	}
 	// If there is an exact height query, return the result immediately
 	// (if it exists).
-	height, ok := lookForHeight(conditions)
-	if ok {
+	height, ok, heightIdx := lookForHeight(conditions)
+
+	// If we have additional constraints and want to query per event
+	// attributed, we cannot simply return all blocks for a height.
+	// But we remember the height we want to find and forward it to
+	// match()
+	if ok && (!matchEvents || (matchEvents && len(conditions) == 2)) {
```

Comment:
Please add to the comment that the check on the length of conditions being 2 is actually checking that there is no extra condition (2 = height + matchEvents)
... unless I'm getting it wrong

Reply:
That's correct. Done.